components/commands/common/BUILD.gn

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2023 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import("//mojo/public/tools/bindings/mojom.gni")
+
+static_library("common") {
```

Comment:
We recommend to use `component` for new target - https://bravesoftware.slack.com/archives/C7VLGSR55/p1673461442230819

Reply:
I've just been trying this but I've been struggling a bit - after converting it to a `component` target I'm getting a bunch of undefined symbol errors, as if it stops including the `.cc` files. Do you have any suggestions (my gn-fu is weak :cry: )

Reply:
You may need to export what you want to use from outside this component.

1. Make COMMAND_EXPORT from a export file - ex) https://source.chromium.org/chromium/chromium/src/+/main:ui/views/views_export.h?q=views_export.h&ss=chromium  - replace VIEWS_EXPORT with COMMAND_EXPORT
2. Mark symbols to be exported, e.g. COMMAND_EXPORT void FooFucntion(); or class COMMAND_EXPORT BarClass ...

Reply:
I'm not sure if COMMAND_COMMON / COMMAND_BROWSER would be better. I believe @goodov could give us advice on this!

Reply:
Please refer to this article: https://chromium.googlesource.com/chromium/src/+/main/docs/component_build.md

The parts you're missing are in this section:
https://chromium.googlesource.com/chromium/src/+/main/docs/component_build.md#exporting-and-importing-symbols

You component name should be "commands_common" (follow the directory structure). Key takeaways from the article that you should apply:
1. add `output_name = "commands_common"`, `defines = [ "IS_COMMANDS_COMMON_IMPL" ]` to your gn target.
2. make use of `base/component_export.h` by marking all required parts with `COMPONENT_EXPORT(COMMANDS_COMMON)`

Please don't mimic `views_export.h` as this the old approach before `base/component_export.h` existed.

Reply:
Okay, the bit I was missing was the convention for the export name - I was trying to use `brave_commands`/`COMPONENT_EXPORT(BRAVE_COMMANDS)`/`IS_BRAVE_COMMANDS_IMPL`. Thanks everyone!